### ISSUE TITLE
TH2 Underpowered/TH3 Overpowered

### ISSUE THREAD

**Issue by pnavuf** [EdenServer/community#1275]
Based on my observations and that of a friend's, there is a large discrepancy between TH2 and TH3 drop rates. When farming Archer's Rings from Stroper Chymes in Ordelle Caves, TH3 obtained 2-3x or better drop rates. We have each farmed 18-20 rings but my friend was able to do it in 1/3rd the number of kills. According to FFXIclopedia, TH3 should be a minuscule increase to drop rate (on the order of +1%) while TH2 should be massive (effectively +50%). This is not a statistical abnormality or getting unlucky, this is night and day. It may be that TH3 is applying a flat percentage increase to drop rates before TH2 which would make this issue much more noticeable on low droprate items but there is no way to know that without checking the code.

Furthermore, the issue is compounded by the fact that Thief's Knife drop rate is very small on this server. Effectively, only Kupo players have this item. This exacerbates the gap between new and old players. Thief's Knife is currently selling for 35 million gil. Other players likely know of this issue but have not reported it in order to reap the benefits of the bug.

**Comment by godmode** [comment]
On items with low drop rates such as Archer Rings TH3/TH4 can provide an extremely potent increase to Treasure Hunter. On items with high drop rate such as Lizard Tails TH3 is a negligible increase from TH2. Based on your data I believe this is not a bug and is working as intended but I’m leaving it open for awhile for others to chime in. Just in case I’m missing something.

**Comment by gweivyth** [comment]
I haven't noticed a wide margin between TH2 and TH3 on most things, mostly just what Godmode said.  I saw a pretty significant drop in Serpentine guns in Davoi, but everything else was still pretty close.  (This was back before that camp got nerfed obviously.)  

I used to have a Thief's Knife, ended up giving it back when the player I was borrowing it from came back to Eden.  I probably spent about 30~ hours farming that area both with and without the Knife.

**Comment by pnavuf** [comment]
Is there any evidence for low drop rate items being more affected by TH3/TH4? I feel like this something people would have discovered during retail if it were the case. All the testing on TH I have found centers on common drop rate items to get large sample sizes. A change of this magnitude should be backed by evidence. Right now, the game is being changed because of the way TH3 works. If TH3 is being kept the same, then Thief's Knife should be brought up to it's March 2007 drop rate (100%) so that Kupo players do not have an unfair advantage.

**Comment by gweivyth** [comment]
While I don't have the same kind of insight into how TH works on Eden like Godmode does, I do know that how it worked on base DSP was that it provided extra rolls a percentage of the time based on your treasure hunter level.  So normal drops drop on a hit/miss system based on their drop rate %.  If you "missed" that roll and didn't have Treasure Hunter at all, it would just not drop.  If you missed and you DID have TH1, it would roll a % of the time (this was a variable you could change, so you could set TH levels to do this however often you wanted to) to see if you would get another shot at rolling for that miss/fail.  I'm not sure if anyone knows how it worked exactly on retail, but I think this was DSP's best guess. 

As for the second point, why would Kupo players have some kind of advantage?  It wasn't 100% drop on Kupo at any point.  And just being a Kupo player doesn't necessarily mean you have a Thief's Knife, I don't have one and there are a few people who have leveled THF after Kupo died that don't have it either.

**Comment by Lilswitche** [comment]
Ref changing Sozu Rogberry: Discussion/opinions for development related topics should go to to the forums for more public consumption.

Ref Archer's Rings: Your numbers for rings/kills are still a small sample size but +1 items will have a large relative increase for drop rates that are excessively low.  For example, if you were to take something with a 0.5% drop rate and bring it up to 1.5%, that would be +200% drop rate increase vs something like 50% to 51% representing +2% increase.

**Comment by jeffmcruz** [comment]
So, listed on bgwiki, it has how TH affected drop rates. TH1/2 gave a chance(50%) at an additional roll for loot. If you won the roll, that would increase an items drop rate by 25% of its drop rate. TH3/4 gave a flat 1% chance to get an item. With archers ring(which has a 2% drop rate), th1/2, would give an additional chance at adding .5% each at the item dropping. Th3/4 giving a flat 1%, doubles the effectiveness of TH 1 and TH 2.

**Comment by gweivyth** [comment]
> With archers ring(which has a 2% drop rate), th1/2, would give an additional chance at adding .5% each at the item dropping. Th3/4 giving a flat 1%, doubles the effectiveness of TH 1 and TH 2.

I don't believe treasure hunter is cumulative, I think it just takes the highest level of TH you have.

**Comment by Elysien** [comment]
> Is there any evidence for low drop rate items being more affected by TH3/TH4? I feel like this something people would have discovered during retail if it were the case. All the testing on TH I have found centers on common drop rate items to get large sample sizes. A change of this magnitude should be backed by evidence. Right now, the game is being changed because of the way TH3 works. If TH3 is being kept the same, then Thief's Knife should be brought up to it's March 2007 drop rate (100%) so that Kupo players do not have an unfair advantage.

Simple mathematics, but first understand how Treasure Hunter I and II work (I assume it's coded this way on Eden, but I can't be sure--in anycase, I'm sure, based on what Godmode said, that it is reminiscent of it). 

A monster has a item with a 30% drop rate, you have TH2. It doesn't drop, but TH1 will kick in, and reroll another 30% chance. That reroll fails, TH2 kicks in and, yet again, rerolls another 30%. TH3 (and TH4) is thought to work differently, it's literally TH+1, as in, adds a 1% chance to the reroll. So, using the aforementioned, the drop rate is 30%, it doesn't drop, the following rerolls are now 31%. (If you had TH4, this would be 32%.

As you can see, the increase is very minimal here (3%), because the returns are significantly diminished the higher the items natural drop rate is.

Let's say Archer's Ring has a 1% drop rate. With TH2, it effectively becomes a 2.5% drop. Now, apply TH3 and it now becomes a 3.5% drop, might not look it but this is significant, it's a 40% increase.

**Comment by MGramolini** [comment]
![image](https://user-images.githubusercontent.com/1774728/54379415-bd1ad300-465f-11e9-8342-115f2058350d.png)

I believe Eden's TH1 and TH2 to add a 50% chance additional roll  each. In this case you see that it's likely a 2% drop with no TH (between ffxidb.com and the wiki both being close to this value). Below you will see 0.99 because this is 50% chance of the original roll (1%).

Without TH that would yield us: 2%
With TH1 that would be: 1 - (0.98 * 0.99) = 2.98%
With TH2 that would be: 1 - (0.98 * 0.99 * 0.99) = 3.95%
With TH3 that would be: 1 - (0.97 * 0.98 * 0.98) = 6.84%
With TH4 that will be: 1- (0.96 * 0.97 * 0.97) = 9.67%

**Comment by gweivyth** [comment]
Weird, I was under the impression that your % shot at a reroll raised as your TH level raised.  So it worked in tiers of like 25% chance at TH1, 35% at TH2, etc., and it took the highest % and applied it once.  My code knowledge is pretty elementary admittedly but that's at least what it looked like last time I looked at it on base DSP. 

If it works as described here that's a lot more potent than I gave it credit for, wow.
